File: CPP/Best_Time_To_Buy_And_Sell.cpp
#include <iostream>
#include <stdio.h>
#include <vector>

using namespace std;
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int largestDiff = 0;
        int minVal = 10001;
        int n = prices.size();
        if(n == 1){
          return largestDiff;
        }
        for (int i = 0; i<n; i++){
            if (prices[i] < minVal){
                minVal = prices[i];
            }
            else if ((prices[i]-minVal) > largestDiff){
                largestDiff = prices[i] - minVal;
            }
            else{
              continue;
            }
        }
        return largestDiff;
    }
};

int main(){
    Solution s;
    vector<int> prices {7,1,5,3,6,4};
    int ans = s.maxProfit(prices);
    cout<<ans;
}

File: Python/Best time to buy and sell.py
class Solution:
    def maxProfit(self, prices):
        min_price = float('inf')
        max_profit = 0
        for i in range(len(prices)):
          if prices[i] < min_price:
            min_price = prices[i]
          else:
            max_profit = max(prices[i]-min_price, max_profit)
        return max_profit
    
s = Solution()
prices = [7,1,5,3,6,4]
print(s.maxProfit(prices))